string/manipulate.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @typedef {{
+ *      string: string,
+ * }|Object} mojave.replacementStringObject
+ */
+
+
+/**
+ * Replaces all occurrences of a given matching string with a given replacement string
+ *
+ * @param {string}                         text
+ * @param {mojave.replacementStringObject} replacements
```

Comment:
Just use `Object<string,string>` as type, please.

Reply:
Sure, thats way easier to read anyway.